src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InMemoryStorage.cs

```diff
@@ -13,10 +14,10 @@ internal partial class DiagnosticIncrementalAnalyzer
         private static class InMemoryStorage
         {
             // the reason using nested map rather than having tuple as key is so that I dont have a gigantic map
-            private static readonly ConcurrentDictionary<DiagnosticAnalyzer, ConcurrentDictionary<(object key, string stateKey), CacheEntry>> s_map =
+            private static readonly ConcurrentDictionary<DiagnosticAnalyzer, ConcurrentDictionary<(object key, (Type, string) stateKey), CacheEntry>> s_map =
```

Comment:
we used to actually dump this info to disk.  so we needed these full string names that were unique to use in teh db.  there's no need for this since this is all now in-memory.  we just need a unique key, which we can get here.

note: looking at this now, given that this is already keyed off DiagnosticAnalyzer, the type portion is redundant.  I'm going to remove that.